cap/modules/services/views/adl.py

```diff
@@ -0,0 +1,344 @@
+
+
+def _table_block_parser(block):
+    block_data = {
+        'data': []
+    }
+    for line in block.lines:
+        if line.text == TABLE_HEADER:
+            # Todo: Make check more general since header can be different.
+            continue
+        values = line.text.split()
+        if len(values) == 5:
+            block_data['data'].append({
+                'val': values[0],
+                'err-': values[1],
+                'err+': values[2],
+                'xmin': values[3],
+                'xmax': values[4],
+            })
+        else:
+            line_key_value = _get_key_value(line)
+            if not line_key_value:
+                _adl_logger(
+                    _type='info',
+                    message='Skipping invalid keyword value pair {}'.format(
+                        line))
+                continue
+            block_data.update({
+                line_key_value[0]: line_key_value[-1]
+            })
+    return block_data
+
+
+def _define_block_parser(block):
+    block_data = {}
+    line_data = block.line.text.split("=")
+    _define_key = line_data[0].strip().split(" ")[-1]
+    _define_value = line_data[1].strip()
+    block_data.update({
+        _define_key: _define_value
+    })
+    return block_data
+
+
+def _union_keyword_parser(line_text):
+    block_data = {}
+    if ':' not in line_text:
+        return block_data
+    line_data = line_text.split(":", 1)
+    line_key_value = line_data[0].strip().split(' ', 1)
+    if _regex_block_match(line_key_value[0]):
+        block_data.update({
+            line_key_value[0]: line_key_value[-1]
+        })
+    block_data.update({
+        'parent': line_data[-1].strip()
+    })
+    return block_data
+
+
+def _adl_parser(adl_file=None):
+    """Returns the ADL parsed file as JSON object.
+
+    :param type: File
+    :param adl_file: Input .adl from the user.
+    :return : JSON object
+    """
+    block_data = []
+    json_parsed = {
+        'objects': [],
+        'regions': [],
+        'infos': [],
+        'tables': [],
+        'define': [],
+    }
+
+    with adl_file.storage().open() as file_handler:
+        for b in _collect_blocks(_collect_lines(file_handler)):
+            block_data.append(b)
+
+    for block in block_data:
+        if block.block_type == 'object':
+            json_parsed['objects'].append(_object_block_parser(block))
+        if block.block_type == 'region':
+            json_parsed['regions'].append(_region_block_parser(block))
+        if block.block_type == 'info':
+            json_parsed['infos'].append(_info_block_parser(block))
+        if block.block_type == 'table':
+            json_parsed['tables'].append(_table_block_parser(block))
+        if block.block_type == 'define':
+            json_parsed['define'].append(_define_block_parser(block))
+
+    return json_parsed
+
+
+@blueprint.route('/adl/<record_id>/<file_name>')
+@login_required
+def ingest_adl(record_id, file_name):
```

Comment:
- [ ] keep only views in `views.py` 
- [ ] move all other adl related utility methods to `cap/modules/services/utils/adl.py`
- [ ] add endpoint to parse uploaded ADL file